test/data/EnumerableMap.ts

```diff
@@ -0,0 +1,415 @@
+      [deployer] = await ethers.getSigners();
+      instance = await new EnumerableMapAddressToAddressMock__factory(
+        deployer,
+      ).deploy();
+    });
+
+    describe('__internal', () => {
+      const addressOne = bnToAddress(BigNumber.from('100'));
+      const addressTwo = bnToAddress(BigNumber.from('200'));
+      const addressThree = bnToAddress(BigNumber.from('300'));
+      const addressFour = bnToAddress(BigNumber.from('400'));
+      const addressFive = bnToAddress(BigNumber.from('500'));
+      const addressSix = bnToAddress(BigNumber.from('600'));
+
+      describe('#at(uint256)', () => {
+        it('returns value coresponding to index provided', async () => {
+          await instance['set(address,address)'](addressOne, addressFour);
+          await instance['set(address,address)'](addressTwo, addressFive);
+          await instance['set(address,address)'](addressThree, addressSix);
+
+          expect(await instance['at(uint256)'](0)).to.deep.equal([
+            addressOne,
+            addressFour,
+          ]);
+          expect(await instance['at(uint256)'](1)).to.deep.equal([
+            addressTwo,
+            addressFive,
+          ]);
+          expect(await instance['at(uint256)'](2)).to.deep.equal([
+            addressThree,
+            addressSix,
+          ]);
+        });
+
+        describe('reverts if', () => {
+          it('index is out of bounds', async () => {
+            await expect(
+              instance['at(uint256)'](0),
+            ).to.be.revertedWithCustomError(
+              instance,
+              'EnumerableMap__IndexOutOfBounds',
+            );
+          });
+        });
+      });
+
+      describe('#contains(address)', () => {
+        it('returns true if value has been added', async () => {
+          await instance['set(address,address)'](addressOne, addressFour);
+          await instance['set(address,address)'](addressTwo, addressFive);
+          await instance['set(address,address)'](addressThree, addressSix);
+
+          expect(await instance['contains(address)'](addressOne)).to.be.true;
+          expect(await instance['contains(address)'](addressTwo)).to.be.true;
+          expect(await instance['contains(address)'](addressThree)).to.be.true;
+        });
+
+        it('returns false if value has not been added', async () => {
+          expect(await instance['contains(address)'](addressFour)).to.be.false;
+        });
+      });
+
+      describe('#length()', () => {
+        it('returns length of enumerable map', async () => {
+          expect(await instance['length()']()).to.equal(0);
+
+          await instance['set(address,address)'](addressOne, addressFour);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['set(address,address)'](addressTwo, addressFive);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['set(address,address)'](addressThree, addressSix);
```

Comment:
perhaps you can add a case or logic that shows adding an entry of an existing key does not increase the length

```js
...
await instance['set(address,address)'](addressThree, addressSix);
expect(await instance['length()']()).to.equal(3);

await instance['set(address,address)'](addressThree, addressSeven);
expect(await instance['length()']()).to.equal(3);
...
```